src/api/schema/components/mod.rs

```diff
@@ -39,11 +40,11 @@ pub enum ComponentKind {
 }
 
 impl Component {
-    fn get_component_id(&self) -> &str {
+    fn get_component_id(&self) -> String {
         match self {
-            Component::Source(c) => c.0.component_id.as_str(),
-            Component::Transform(c) => c.0.component_id.as_str(),
-            Component::Sink(c) => c.0.component_id.as_str(),
+            Component::Source(c) => c.0.component_id.to_string(),
+            Component::Transform(c) => c.0.component_id.to_string(),
+            Component::Sink(c) => c.0.component_id.to_string(),
```

Comment:
Nit: Since `ComponentId.id` is public, we could avoid the allocation with:

```rust
fn get_component_id(&self) -> &str {
    match self {
        Component::Source(c) => c.0.component_id.id.as_str(),
        Component::Transform(c) => c.0.component_id.id.as_str(),
        Component::Sink(c) => c.0.component_id.id.as_str(),
    }
}
```

Alternatively, if it's worth hiding the inner `id` for whatever reason, it could follow the `String::as_str()` implementation with:

```rust
impl ComponentId {
    pub fn as_str(&self) -> &str {
        &self.id
    }
}
```

Reply:
This conversion to `&str` works well for now, be if we look ahead and have the following

```rust
let component_id = ComponentId {
    id: "foo".to_string(),
    scope: Scope::Pipeline("bar".to_string()),
};
```
The `component_id.id.as_str()` will probably not be the same than `component_id.to_string()`.
We might have to use the `bar#foo` representation.

Reply:
I think this could be created _once_ at the time of `ComponentId` construction, and then `.as_str()` would reference that 'cached' version. That'd make allocation a one-time cost.

In either case, it'd remain an implementation detail -- the caller doesn't need to consider what `.as_str()` returns.

What do you think?

Reply:
That seems fair to me to have both `inner: String` and `id: String`. I don't really know if having one more `String` in memory or doing one more allocation is better though.

Reply:
I'd personally opt for `.as_str()`, and create a `ComponentId::new()` method to hide the implementation details of a later `inner` cache.

That would avoid unnecessary allocations for now, when just an `id` is present.

And in future, that would reduce the cost of formatting to a single allocation (at least where taking ownership of it is unnecessary.)

It seems like in most code paths, at least one call to `.to_string()` is present, so having this cost at the time of creation vs. later doesn't seem any more expensive. And it's less expensive in any code path where we're taking the reference to the string twice -- which seems often enough for this to be worth it.

It's minor overall, but I think it's worth doing for high-volume API queries where those string allocations can add up (filtering, etc.)